code/modules/cooking/machinery/foodcart.dm

```diff
@@ -0,0 +1,147 @@
+	QDEL_NULL(cart_griddle)
+	QDEL_NULL(cart_smartfridge)
+	QDEL_NULL(cart_table)
+	QDEL_NULL(cart_tent)
+	packed_things.Cut()
+	return ..()
+
+/obj/machinery/food_cart/get_examine_text(mob/user, distance, is_adjacent, infix, suffix)
+	. = ..()
+	if(stat & BROKEN)
+		return
+	if(cart_griddle.stat & BROKEN)
+		. += SPAN_WARNING("The stand's [SPAN_BOLD("grill")] is completely broken!")
+	else
+		. += SPAN_NOTICE("The stand's [SPAN_BOLD("grill")] is intact.")
+	. += SPAN_NOTICE("The stand's [SPAN_BOLD("fridge")] seems fine.") //weirdly enough, these fridges don't break
+	. += SPAN_NOTICE("The stand's [SPAN_BOLD("table")] seems fine.")
+
+/obj/machinery/food_cart/proc/pack_up() // Retract the structures into the food cart
+	if(!unpacked)
+		return
+	visible_message(SPAN_NOTICE("[src] retracts all of it's unpacked components."))
+	for(var/obj/object as anything in packed_things)
+		UnregisterSignal(object, COMSIG_MOVABLE_MOVED)
+		object.forceMove(src)
+	if(!(cart_griddle?.stat & BROKEN)) // Don't draw power if it's packed inside
+		cart_griddle?.stat = POWEROFF
+		cart_griddle?.use_power = POWER_USE_OFF
+		cart_griddle?.update_icon()
+	if(!(cart_smartfridge?.stat & BROKEN))
+		cart_smartfridge.use_power = POWER_USE_OFF
+	anchored = FALSE
+	unpacked = FALSE
+
+/obj/machinery/food_cart/proc/unpack(mob/user) // Deploy the structures inside the food cart
+	if(unpacked)
+		return
+	if(!check_setup_place())
+		to_chat(user, SPAN_WARNING("There isn't enough room to unpack here! Bad spaces were marked in red."))
+		return
+	visible_message(SPAN_NOTICE("[src] expands into a full stand."))
+	anchored = TRUE
+	if(!(cart_griddle?.stat & BROKEN))
+		cart_griddle?.stat = POWEROFF
+	if(!(cart_smartfridge?.stat & BROKEN)) // Unpacked, draw power
+		cart_smartfridge.use_power = POWER_USE_IDLE
+	var/iteration = 1
+	var/turf/grabbed_turf = get_step(get_turf(src), EAST)
+	for(var/angle in list(0, -45, -45, 45))
+		var/turf/T = get_step(grabbed_turf, turn(SOUTH, angle))
+		var/obj/thing = packed_things[iteration]
+		thing.forceMove(T)
+		RegisterSignal(thing, COMSIG_MOVABLE_MOVED, PROC_REF(lost_part))
+		iteration++
+	unpacked = TRUE
+
+/obj/machinery/food_cart/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(stat & BROKEN)
+		to_chat(user, SPAN_WARNING("[src] is completely busted."))
+		return
+	var/obj/item/card/id/id_card = user.GetIdCard()
+	if(!check_access(id_card))
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+		return
+	playsound(get_turf(src), 'sound/items/rfd_start.ogg', 50, FALSE)
+	to_chat(user, SPAN_NOTICE("You attempt to [unpacked ? "pack up" :"unpack"] [src]..."))
+	if(!do_after(user, 5 SECONDS, src))
+		playsound(get_turf(src), 'sound/items/rfd_interrupt.ogg', 50, FALSE)
+		return
+	playsound(get_turf(src), 'sound/items/rfd_end.ogg', 50, FALSE)
+	if(unpacked)
+		pack_up()
+	else
+		unpack(user)
+
+/obj/machinery/food_cart/proc/check_setup_place() // Step and check whether cart structures have non-dense space to placed onto
```

Comment:
https://github.com/Aurorastation/Aurora.3/blob/master/.github/guides/STYLE.md#documenting-a-proc

That isn't a DMDoc